### PR TITLE
Remove BATCH_SIZE from CMake and preprocessor

### PR DESCRIPTION
**Pull request by garth-wells**

### CODE REVIEW

**Comment by IgorBaratta** [comment]
I've made small changes to the PR.

Why is the elasticity reference disabled?

File: src/main.cpp [hunk 1]
```diff
@@ -45,49 +42,79 @@ int main(int argc, char** argv)
...
-      else if (operator_str == "elasticity")
-      {
-        if (scalar_type_str == "float")
-          reference::elasticityX::run<float, BATCH_SIZE>(degree, Ndofs);
+        {
+          if (batch_size == 1)
+            reference::stiffness::run<float, 1>(degree, Ndofs);
+          else if (batch_size == 4)
+            reference::stiffness::run<float, 4>(degree, Ndofs);
+          else if (batch_size == 8)
```
Comment (IgorBaratta) [inline src/main.cpp:94]: Add float 16?

**Comment by garth-wells** [comment]
> I've made small changes to the PR.
> 
> Why is the elasticity reference disabled?

Re-enabled now. I didn't bother at first because I saw in another PR that you removed elasticity.

File: src/main.cpp (hunk 1 above)
Comment (garth-wells) [reply src/main.cpp:92]: SIMD size for floats fixed.

**Comment by IgorBaratta** [comment]
> Re-enabled now. I didn't bother at first because I saw in another PR that you removed elasticity.

I'll rebase my branch on this.

**Pull request merged**